README.md

```diff
@@ -1,178 +1,114 @@
-      /- SampleBot1
-        /- bot1.bot    (bot project file, maybe we can call it .botproject, all files should be referenced here)
+      /- Node
     /- Composer
       /- config.json   (the config to the composer)
-      /- launcher.json (composer use this config to start Bot Launcher)
+    ...somewhere else on disk
+    /-echo-bot
+      /.bot
 
+## a bot's .bot file
 
-It's very clearly divided in the top level, each component has it's own folder. 
+Here is an example of a .bot file intended to run on Node, with glob patterns for bot asset discovery.
 
-## .bot file
-The most important file in `Bots` folder is each .bot file, which defined a bot, containing all the references within this bot. 
 ```
-/Bots/SampleBot1/bot1.bot
-
 {
-    services: [
-        // luis
-        // qna
-    ],
-
-    files: [
-        ./**/*.lu
-        ./**/*.lg
-        ./**/*.dialog
-    ]
-    
-    entry: "./dialogs/start.dialog"
+  services: [
+    {
+      type: "luis",
+      endpoint: ${luis_endpoint},
+      app_id: ${luis_app_id},
+      token: ${luis_token} 
+    },
+    {
+      type: "qna",
+      endpoint: ${qna_endpoint},
+      app_id: ${qna_app_id},
+      token: ${qna_token} 
+    }
+  ],
 
-}
+  files: [
+    "./**/*.lu",
+    "./**/*.lg",
+    "./**/*.dialog"
+  ]
 
+  dialogEntry: "./dialogs/start.dialog",
+  start: "node index.js"
+}
 ```
 
-.bot is basicly a project file, include all the references to all assets, and other necessary infomation to run a bot. 
-
-## Bot launcher configuration
+the .bot file is used for the following:
+- asset discovery via glob patterns
+- service discovery for luis and qna, etc
+- service credentials
+- entry points for the dialog tree and the bot's runtime.
 
-Since all assets are defined in bots. The bot launcher's config is simple, only need to know about where to find a bot.
-
-```
-/BotLauncher/CSharp/appsettings.json
-{
-    // this is the bot to be launched
-    bot: {
-        provider: "localDisk",
-        path: "../../Bots/SampleBot1/bot1.bot
-    }
-
-    // it may include some othe info to run like
-    endpoint: "localhost:3979"
-}
-```
+**note:** a goal to consider is the service credentials being *templated* in this file. we've made assumptions in the past about where these credentials should be stored and their shape.
 
 ## Composer configuration
 
 ### config.json
 
-config.json in composer is where to specfic 
-1. which bot we are editing (if we follow vscode convention, this should be in .composer, but let's make it simple here)
-2. which launcher we are using
-
 ```
-/Composer/config.json 
 {
-    // this is the bot we are editing\composing
-    bot: {
-        provider: "localDisk"
-        path: "../../Bots/SampleBot1/bot1.bot"
-    }
-    
-    launcher: {
-        startCommand: "cp launcher.json '../../BotLauncher/CSharp/appsetings.json' && cd '../../BotLauncher/CSharp' && dotnet run"
+  // the path to the BotLauncher root
+  launcherDir: "../../BotLauncher",
+  bots: [
```

Comment:
why can't this be an array of bots?

Reply:
this is just saying the current editing bot, because we assume you are editing only one bot each time. you can store the history of bots or the path of folder where you usually you are looking for bots. Usually such thing is stored in another place.